Lib/test/datetimetester.py

```diff
@@ -57,6 +57,33 @@
 NAN = float("nan")
 
 
+def _utcfromtimestamp(klass, ts):
+    """Simple re-implementation of datetime.utcfromtimestamp.
+
+    utcfromtimestamp is deprecated because it returns a naïve datetime object
+    despite being aware that it is UTC. This sort of deliberately wrong object
+    happens to be useful when calculating transition times from a TZif file,
+    so this is a re-implementation of that.
+    """
+    ats = abs(ts)
+    sign = int(ts / ats)
+
+    ts, frac = divmod(ats, 1.)
+    ts *= sign
```

Comment:
We can use math.copysign here for a more efficient implementation.

Reply:
Yeah, this whole thing was an effort to avoid importing `math`, so it's something of a re-implementation of `math.modf`. At the end of the day that seems... pointless, particularly for the test suite, so I just went ahead and used `math.modf`.